src/ts-node.ts

```diff
@@ -61,6 +60,8 @@ function registerTSNode(root: string) {
 function loadTSConfig(root: string): TSConfig | undefined {
   const tsconfigPath = path.join(root, 'tsconfig.json')
   if (fs.existsSync(tsconfigPath)) {
+    // Lazy-require 'typescript' to support JavaScript @oclif projects.
+    const {parseConfigFileTextToJson} = require('typescript')
```

Comment:
I think this doesn't solve the root of the issue where `typescript` is currently listed in `devDependencies` instead of `dependencies` in the package file. I suspect this will still break for TS projects.

I'm only here since our project is affected as well, so 🙇 thank you for taking the time to submit a PR!

Reply:
@trevor-scheer yes, that's why I updated 6042e18 to be even lazier about it.

Reply:
Thanks for giving it a look over though, glad to know we're not the only ones getting bitten by it 🙏

Reply:
What I'm suggesting is, now that `typescript` is being required as an actual runtime dependency (no matter how lazily), it needs to be listed in `dependencies` instead of `devDependencies` for this package.